src/nvim/diff.c

```diff
@@ -952,7 +952,7 @@ void ex_diffpatch(exarg_T *eap)
   } else {
     if (curbuf->b_fname != NULL) {
       newname = vim_strnsave(curbuf->b_fname,
-                             (int)(STRLEN(curbuf->b_fname) + 4));
+                             (size_t)(STRLEN(curbuf->b_fname) + 4));
```

Comment:
Is this cast needed? `strlen()` returns a `size_t` and I don't think `+ 4` will make it signed.